### ISSUE TITLE
Add support for Operator Autopilot API

### ISSUE THREAD

**Issue by rhughes1** [WeAreFarmGeek/diplomat#206]
I've been trying to look through the documentation to find a way to pull the health of the cluster using the [Autopilot API](https://www.consul.io/api/operator/autopilot.html). In particular I'm looking to see if the nodes are voters or not.

Below would be an example of getting the current Autopilot configuration:

Below would be an example of what I was thinking it would look like for getting the health, configuration and updating the configuration:

```ruby
Diplomat::Autopilot.get_health()
```

```ruby
Diplomat::Autopilot.get_configuration()
```

```ruby
Diplomat::Autopilot.update(value)
```

I'm curious if there's some workaround that I'm not aware of to get the voter status in the cluster?

**Comment by pierresouchay** [comment]
The API `/v1/agent/members` aka Diplomat::Members.get() also give you this information, with the tag `nonvoter": "1",`.

But if you want to do a PR to add support for Autopilot API feel free :)

**Comment by pierresouchay** [comment]
@rhughes1 Did you find your way to implement what you wanted?

**Comment by rhughes1** [comment]
@pierresouchay believe I did... 

Although I think I could open a PR for an autopilot endpoint. My only concern is being able to make valid unit tests :s

**Comment by pierresouchay** [comment]
@rhughes1 If you don't know how to add unit test, try submitting it, we will help you write the unit tests

**Comment by pierresouchay** [comment]
@rhughes1 Merge upstream and released as version 2.4.0 (will be available in rubygems in a few minutes).

If you want to add support for more endpoints of [Autopilot API](https://www.consul.io/api/operator/autopilot.html), feel free.

Thanks a lot for your contribution!
